stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1234,3 +1250,405 @@ fn balances_from_keys(
+                Value::UInt(100),
+                pox_addr,
+                Value::UInt(block_height as u128),
+                Value::UInt(lock_period),
+                signer_key_val.clone(),
+            ],
+        ),
+    ];
+    // (define-public (delegate-stack-stx (stacker principal)
+    //                                (amount-ustx uint)
+    //                                (pox-addr { version: (buff 1), hashbytes: (buff 32) })
+    //                                (start-burn-ht uint)
+    //                                (lock-period uint)
+    //                                (signer-key (buff 33)))
+
+    let latest_block = peer.tenure_with_txs(&txs, &mut coinbase_nonce);
+    let delegation_state = get_delegation_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(stacker_key).to_account_principal(),
+    )
+    .expect("No delegation state, delegate-stx failed")
+    .expect_tuple();
+
+    let state_signer_key_optional = delegation_state.get("signer-key").unwrap();
+    assert_eq!(
+        state_signer_key_optional.to_string(),
+        Value::Optional(OptionalData {
+            data: Some(Box::new(signer_key_val.clone()))
+        })
+        .to_string()
+    );
+
+    let stacking_state = get_stacking_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(stacker_key).to_account_principal(),
+    )
+    .expect("No stacking state, stack-stx failed")
+    .expect_tuple();
+
+    let state_signer_key = stacking_state.get("signer-key").unwrap();
+    assert_eq!(state_signer_key.to_string(), signer_key_val.to_string());
+}
+
+pub fn get_stacking_state_pox_4(
+    peer: &mut TestPeer,
+    tip: &StacksBlockId,
+    account: &PrincipalData,
+) -> Option<Value> {
+    with_clarity_db_ro(peer, tip, |db| {
+        let lookup_tuple = Value::Tuple(
+            TupleData::from_data(vec![("stacker".into(), account.clone().into())]).unwrap(),
+        );
+        let epoch = db.get_clarity_epoch_version();
+        db.fetch_entry_unknown_descriptor(
+            &boot_code_id(boot::POX_4_NAME, false),
+            "stacking-state",
+            &lookup_tuple,
+            &epoch,
+        )
+        .unwrap()
+        .expect_optional()
+    })
+}
+
+pub fn get_delegation_state_pox_4(
+    peer: &mut TestPeer,
+    tip: &StacksBlockId,
+    account: &PrincipalData,
+) -> Option<Value> {
+    with_clarity_db_ro(peer, tip, |db| {
+        let lookup_tuple = Value::Tuple(
+            TupleData::from_data(vec![("stacker".into(), account.clone().into())]).unwrap(),
+        );
+        let epoch = db.get_clarity_epoch_version();
+        db.fetch_entry_unknown_descriptor(
+            &boot_code_id(boot::POX_4_NAME, false),
+            "delegation-state",
+            &lookup_tuple,
+            &epoch,
+        )
+        .unwrap()
+        .expect_optional()
+    })
+}
+
+pub fn with_clarity_db_ro<F, R>(peer: &mut TestPeer, tip: &StacksBlockId, todo: F) -> R
```

Comment:
Pretty sure this is a dup of code that lives elsewhere.

Reply:
Correct it's from `pox_2_tests.rs`. Felt a bit strange to import from test files targeting previous pox versions. Can I pull all helper functions into a new separate file `pox_test_helpers.rs`?